### ISSUE TITLE
feature: task compartamentalization

### ISSUE THREAD

**Issue by kernelkind** [Medvedev91/timeto.me#86]
I would like tasks to be tied to a specific activity. So when I create a task in my "work" activity and then switch to a different activity, I don't see that task anymore. But when I go back to "work", I see my work tasks. 

What are your thoughts on this feature request?

**Comment by kernelkind** [comment]
@Medvedev91 if you like this feature request, I'll begin work for a PR

**Comment by Medvedev91** [comment]
Do you mean, while the "Work" activity is running, you want to see only "Work" tasks on the home screen?

**Comment by kernelkind** [comment]
> Do you mean, while the "Work" activity is running, you want to see only "Work" tasks on the home screen?

Yes, exactly.

Also, maybe at task creation time assigning the activity to the task could be optional so if users don't necessarily want to make that decision yet they can postpone it to when they want to work on the task. But then you'd have to have a section for the activity-specific tasks and "all" tasks.

Here's an example:
```
Work Tasks:
- answer emails
- feature X

Uncategorized Tasks:
- grocery shopping
- book doctors apt
```

This UX is less clean IMO and it doesn't matter either way to me whether we enforce activity categorization at task creation time or not. I just want to be able to have the option of categorizing the task without having to start it, and I want the task to be hidden until that activity is active.

**Comment by Medvedev91** [comment]
At the moment you can do categorization via task folders.

**Comment by kernelkind** [comment]
> At the moment you can do categorization via task folders.

I like task folders, and I find them useful. But they don't offer the type of compartmentalization that I'm looking for. 

For comparison, the `TMRW` folder filters all tasks for only ones that have tomorrow's date assigned. While I want ability to filter like this folder, I would like a view that only shows tasks that are assigned to the same activity as the activity that is active currently. 

# Here's an example use case for the feature I desire
Let's say I have two activities: `Work` and `Exercise`.

- The `Work` activity is active. 
- I add the following task: `review emails`
- when I click the 'Save' button, a popup appears: `Where do you want to save this task? Work, Exercise, or Uncategorized?`
- I click on `Work`
- on the home screen, the task `review emails` appears under a collapsible heading `Work`
- I add another task: `go for run`
- when I click the 'Save' button, a popup appears: `Where do you want to save this task? Work, Exercise, or Uncategorized?`
- I select `Uncategorized`
- on the home screen, underneath the collapsible heading `Work`, is another collapsible heading named `Uncategorized Tasks`. Underneath that heading is the task `go for run`
- I click on `go for run`
- I am prompted to select an activity for the task, exactly like what would happen in the real app
- I select `Exercise`
- The activity changes to `Exercise` and I see two collapsible headers: `Exercise`, and `Uncategorized Tasks`. **Both of which are empty**
- After my run, I switch back to the `Work` activity. I see two collapsible headers, `Work` and `Uncategorized Tasks`. Under the `Work` header, I see the task `review emails`, and nothing under `Uncategorized Tasks`
- I select `review emails` and **the task is immediately started without prompting for activity type**

Like I said earlier, we're basically giving the user the option to categorize their task at 'task creation time' rather than only at task start time. And if the user wants to wait to categorize the task until task start time, they still have that option, as you saw in my example.

Does this example help you understand my feature request?

**Comment by Medvedev91** [comment]
Thank you for the detailed explanation. Did I understand right, you want to have like activity filters for tasks on the homescreen?

**Comment by kernelkind** [comment]
> Thank you for the detailed explanation. Did I understand right, you want to have like activity filters for tasks on the homescreen?

Yes, exactly. But I only want to see filters for for the current activity and "uncategorized" activities
